components/StatsModal.tsx

```diff
@@ -159,65 +160,38 @@ export default function StatsModal(props: Props) {
   }
 
   async function handleShareImage() {
-    const canvas = document.createElement("canvas");
-    canvas.height = 1400;
-    canvas.width = 900;
-
-    const ctx = canvas.getContext("2d");
-    ctx.fillStyle = resolvedTheme === "dark" ? "#111827" : "#ffffff";
-    ctx.fillRect(0, 0, canvas.width, canvas.height);
-
-    const gap = 10;
-    const paddingH = 200;
-    const paddingT = 400;
-    const size = (canvas.width - paddingH * 2 - gap * 4) / 5;
-
+    const { origin } = absoluteUrl();
+    const answer = decode(game.hash);
     const score =
       game.state.answers[game.state.attempt - 1] === answer
         ? game.state.attempt
         : "X";
-    let text = `Katla ${game.num} ${score}/6\n\n`;
-    ctx.font = "bold 42px sans-serif";
-    ctx.textAlign = "center";
-    ctx.fillStyle = resolvedTheme === "dark" ? "#ffffff" : "#111827";
-    ctx.fillText(text, canvas.width / 2, 300);
-
-    game.state.answers.slice(0, game.state.attempt).forEach((answer, y) => {
-      const states = getAnswerStates(answer, decode(game.hash));
-      states.forEach((state, x) => {
-        if (state === "correct") {
-          ctx.fillStyle = game.state.enableHighContrast ? "#f5793a" : "#15803d";
-        } else if (state === "exist") {
-          ctx.fillStyle = game.state.enableHighContrast ? "#85c0f9" : "#ca8a04";
-        } else {
-          ctx.fillStyle = resolvedTheme === "dark" ? "#374151" : "#6b7280";
-        }
-
-        const marginH = gap * x;
-        const marginV = gap * y;
-        ctx.beginPath();
-        ctx.rect(
-          paddingH + x * size + marginH,
-          paddingT + y * size + marginV,
-          size,
-          size
-        );
-        ctx.fill();
-      });
+    const imageUrl = `${origin}/api/image?gameNumber=${
+      game.num
+    }&answer=${answer}&userAnswer=${game.state.answers.join()}&score=${score}&theme=${resolvedTheme}`;
+
+    await (await fetch(imageUrl)).blob().then((blob) => {
+      if (useNativeShare && navigator.canShare) {
+        const shareData = {
+          files: [
+            new File([blob], "katla.png", {
+              type: "image/jpeg",
+              lastModified: new Date().getTime(),
+            }),
+          ],
+        };
+        navigator.share(shareData).catch(() => {});
+      } else {
```

Comment:
for sobat android & desktop still can share image but it just download the image